ray-operator/controllers/ray/rayjob_controller_test.go

```diff
@@ -192,14 +196,46 @@ var _ = Context("Inside the default namespace", func() {
 				time.Second*3, time.Millisecond*500).Should(BeNil(), "My myRayCluster  = %v", myRayCluster.Name)
 		})
 
-		It("should create more than 1 worker", func() {
-			filterLabels := client.MatchingLabels{common.RayClusterLabelKey: myRayJob.Status.RayClusterName, common.RayNodeGroupLabelKey: "small-group"}
+		It("Should create a number of workers equal to the replica setting", func() {
+			filterLabels := client.MatchingLabels{common.RayClusterLabelKey: myRayJob.Status.RayClusterName, common.RayNodeGroupLabelKey: myRayJob.Spec.RayClusterSpec.WorkerGroupSpecs[0].GroupName}
 			workerPods := corev1.PodList{}
 			Eventually(
 				listResourceFunc(ctx, &workerPods, filterLabels, &client.ListOptions{Namespace: "default"}),
-				time.Second*15, time.Millisecond*500).Should(Equal(3), fmt.Sprintf("workerGroup %v", workerPods.Items))
-			if len(workerPods.Items) > 0 {
-				Expect(workerPods.Items[0].Status.Phase).Should(Or(Equal(corev1.PodRunning), Equal(corev1.PodPending)))
+				time.Second*15, time.Millisecond*500).Should(Equal(int(*myRayJob.Spec.RayClusterSpec.WorkerGroupSpecs[0].Replicas)), fmt.Sprintf("workerGroup %v", workerPods.Items))
+			for _, workerPod := range workerPods.Items {
+				Expect(workerPod.Status.Phase).Should(Or(Equal(corev1.PodRunning), Equal(corev1.PodPending)))
```

Comment:
The document [envtest](https://book.kubebuilder.io/reference/envtest.html) indicates that envtest will start a Kubernetes cluster with etcd and the Kubernetes API server, without kubelet, controller-manager or other components. Hence, there is no built-in Kubernetes Pod controller in the fake cluster. In my understanding, the status will not be updated to `corev1.PodRunning` forever. Can you check this?

Reply:
Thank you for pointing out! The status will not be updated to `corev1.PodRunning` forever. Let me change the logic here.

Reply:
I will remove this test logic here as the status will always be `pending` unless manually updated.